### PR TITLE
MAGN-8410 Improve vertical node type labels in tree view

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

This PR addresses [MAGN-8410](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8410).  
For this PR,
1. Font Awesome for WPF was installed from Nuget Package manager.

This PR uses the fontawesome. All the labels, create, action, query were replaced by images.

icon1
icon3

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR 
- [ ] The level of testing this PR includes is appropriate
  - No new tests required. This is on the View Layer.
- [ ] User facing strings, if any, are extracted into `*.resx` files
  - No new strings added/removed.

### Reviewers

@pboyer

### CODE REVIEW

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 1]
```diff
@@ -36,6 +36,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="FontAwesome.WPF">
+      <HintPath>..\..\extern\FontAwesome\FontAwesome.WPF.dll</HintPath>
+    </Reference>
```
Comment (pboyer) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj]: Cool! But we definitely need a license file.

File: src/DynamoCoreWpf/UI/Converters.cs [hunk 2]
```diff
@@ -2660,5 +2630,89 @@ public object ConvertBack(object value, Type targetType, object parameter, Syste
...
+            public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+            {
+                var grid = value as Grid;
+                
+                //Outerbuttongrid has two different child.
+                if (grid != null && grid.Children.Count > 0)
+                {
+                    //first Child is a border
+                    var child1 = grid.Children[0] as Border;
+                    if (child1 != null)
```
Comment (pboyer) [inline src/DynamoCoreWpf/UI/Converters.cs]: Don't build deeply nested if blocks, just do:

```
if (child1 == null) return x;
```

This is real tough to read.

File: src/DynamoCoreWpf/UI/Converters.cs [hunk 3]
```diff
@@ -2660,5 +2630,89 @@ public object ConvertBack(object value, Type targetType, object parameter, Syste
...
-   }
+        }
+
+        public class ClassViewMarginConverter : IValueConverter
+        {
+            private const int levelmargin = 45;
+            private const int marginTop = -10;
+            private const int marginLeft = -10;
+            private const int viewMarginLeft = -30;
+            private const int factor = 10;
```
Comment (pboyer) [inline src/DynamoCoreWpf/UI/Converters.cs]: http://stackoverflow.com/questions/242534/c-sharp-naming-convention-for-constants

File: test/DynamoCoreWpfTests/ConverterTests.cs [hunk 4]
```diff
@@ -613,27 +613,6 @@ public void FilePathDisplayConverterInternalDoesNotThrow()
...
-            object[] objects = new object[2];
-            objects[0] = headerStrip;
-            objects[1] = listBox;
-            
-            var converter = new TreeViewListBoxTitleConverter();
-            converter.Convert(objects, null, null, null);
-
-            var item = headerItems.FirstOrDefault();
-            Assert.AreEqual("C", item.Text);
-        }
```
Comment (pboyer) [inline test/DynamoCoreWpfTests/ConverterTests.cs]: Sad to remove tests. :(

Comment (ramramps) [inline test/DynamoCoreWpfTests/ConverterTests.cs]: this test is not required. this test is for the create action query labels. We dont have those labels now.

**Comment by ramramps** [comment]
@pboyer - PTAL. btw, how to add the licence file for Fontawesome?

**Comment by pboyer** [comment]
@ramramps There is a file called LICENSE.md that should give many example
of how we display third party license files.

**Comment by ramramps** [comment]
Thanks @pboyer. PTAL.

**Comment by pboyer** [comment]
LGTM.

**Comment by Racel** [comment]
@ramramps - This looks good in Dynamo Sandbox, but I can not verify that is working properly on Dynamo for Revit. Waiting for a fix to MAGN-8474 before I close the YT.

**Pull request merged**